Ginger/GingerCoreNET/Drivers/CoreDrivers/Mobile/Appium/GenericAppiumDriver.cs

```diff
@@ -2234,18 +2279,61 @@ public Byte[] GetScreenshotImage()
 
         public void PerformTap(long x, long y)
         {
-            TapXY(x, y);
+            if (IsRecording)
+            {
+                ElementInfo elemInfo = GetElementAtPoint(x, y).Result;
+                if (elemInfo != null)
+                {
+                    RecordingEventArgs args = new RecordingEventArgs();
+                    args.EventType = eRecordingEvent.ElementRecorded;
+
+                    ElementActionCongifuration configArgs = new ElementActionCongifuration();
+                    configArgs.LocateBy = eLocateBy.ByXPath;
```

Comment:
need to change logic to set the locator by preference:
1- ID
2- Name
3- XPATH
4- X,Y